packages/editor/src/worker/OptimizerWorker.ts

```diff
@@ -0,0 +1,250 @@
+});
+
+/**
+ * Wrapper class for the worker thread. Handles sending and receiving messages to and from the worker.
+ */
+export default class OptimizerWorker {
+  private svgCache: Map<string, HTMLElement> = new Map();
+  private workerInitialized: boolean = false;
+  private sharedMemory: Int8Array = new Int8Array();
+  private running: boolean = false;
+  private onUpdate: OnUpdate = () => {};
+  private onError: OnError = () => {};
+  private onComplete: onComplete = () => {};
+  private pending: Pending | undefined = undefined;
+
+  /* Initializa the worker by declaring a shared array buffer and passing it to the worker. Then wait for confirmation from the worker before setting up anything. */
+  async init() {
+    return new Promise<void>((resolve, reject) => {
+      log.debug("Worker initializing...", worker);
+      const sab = new SharedArrayBuffer(2);
+      this.sharedMemory = new Int8Array(sab);
+      this.request({
+        tag: "Init",
+        sharedMemory: sab,
+      });
+
+      worker.onmessage = async ({ data }: MessageEvent<Resp>) => {
+        if (data.tag === "Ready") {
+          this.workerInitialized = true;
+          log.info("Worker Initialized");
+          this.setup();
+          // if there is a pending request, send it after setup
+          if (this.pending) {
+            log.warn("Worker has pending request, sending...");
+            this.request(this.pending.request);
+            this.onComplete = this.pending.onComplete;
+            this.onError = this.pending.onError;
+            this.onUpdate = this.pending.onUpdate;
+          }
+          resolve();
+        } else {
+          // Shouldn't happen, but ignore other messages
+          console.error(`Unknown Response: ${data}`);
+        }
+      };
+      // Optionally, you can add a timeout to reject the promise if the worker takes too long
+      setTimeout(() => {
+        reject(new Error("Worker initialization timed out"));
+      }, 30000); // 3 seconds timeout, for example
+    });
+  }
+
+  setup() {
+    log.info("Registering worker listeners");
+
+    worker.onerror = (e) => {
+      log.error("Worker error: ", e.message);
+    };
+
+    worker.onmessageerror = (e) => {
+      log.error("Worker message error: ", e);
+    };
+
+    worker.onmessage = async ({ data }: MessageEvent<Resp>) => {
+      log.debug("Received message: ", data);
+      if (data.tag === "Update") {
+        this.onUpdate(optRenderStateToState(data.state, this.svgCache));
+      } else if (data.tag === "Error") {
+        this.onError(data.error);
+      } else if (data.tag === "Ready") {
+        log.info("Worker ready for new input");
+        this.onComplete();
+        if (this.pending) {
+          log.warn("Worker has pending request, sending...");
+          this.request(this.pending.request);
+          this.onComplete = this.pending.onComplete;
+          this.onError = this.pending.onError;
+          this.onUpdate = this.pending.onUpdate;
+        }
```

Comment:
nit: looks like the same logic as worker.onmessage in `init()` above? also, might want to use log.info here instead of log.warn